R/zarr-array.R

```diff
@@ -1317,3 +1262,82 @@ ZarrArray <- R6::R6Class("ZarrArray",
 as.array.ZarrArray = function(x, ...) {
   x$as.array()
 }
+
+get_parallel_settings <- function(on_windows = (.Platform$OS.type == "windows"),
+                                  parallel_option = getOption("pizzarr.parallel_read_enabled", FALSE),
+                                  progress = getOption("pizzarr.progress_bar", FALSE)) {
```

Comment:
Since there are now the 3 different parallel backends (pbapply, future.apply, and parallel), would it make more sense for this option to be a categorical/string instead of a binary flag? The naming of the option suggests that it will control whether the user sees a progress bar during processing (as opposed to something to do with parallelism or the pbapply package), so can we name it something like `parallel_backend` with values `"pbapply"`, `"future.apply"`, and `"parallel"`?

Reply:
I was trying to be minimally invasive so didn't think to do something like this. I'll look at what it.

Reply:
OK, so we have a mess of options / considerations here. I think I like where it stands. 

There are actually only 2 parallel back ends and the wrapper `pbapply`. `pbapply` works with either parallel back end and we call it if we want a progress bar.

`parallel_option` accepts TRUE/FLASE or an integer, or a cluster object, or the string "future". "future" works on any platform but parallel has limited functionality on Windows.

I'll document these and make sure things are clear in options.R.